draft-ietf-tls-tls13.md

```diff
@@ -2242,12 +2267,15 @@ extension type (Supported Group Extension):
 The client MUST supply a "supported_groups" extension containing at
 least one group for each key exchange algorithm (currently
 DHE and ECDHE) for which it offers a cipher suite.
-If the client does not supply a "supported_groups" extension with a
-compatible group, the server MUST NOT negotiate a cipher suite of the
-relevant type.  For instance, if a client supplies only ECDHE groups,
-the server MUST NOT negotiate finite field Diffie-Hellman.  If no
-acceptable group can be selected across all cipher suites, then the
-server MUST generate a fatal "handshake_failure" alert.
+Servers receiving a TLS 1.3 ClientHello without this extension yet
+offering either DHE or ECDHE cipher suites MUST send a
+"missing_extension" alert message and close the connection.
```

Comment:
It's actually quite common to have requirements that something be a certain way and that the other side not check. To give an extreme example, consider the following:
http://rtcweb-wg.github.io/jsep/#sec.profile-names, where you are required
to send UDP/TLS/RTP/SAVP[F] and there is an extensive list of other values you must treat as equivalent to that.

I don't have a problem with a requirement that you MUST NOT negotiate (EC)DHE without a there being a supported_groups extension that contains the group in question and that if there is no valid configuration then you must send "handshake_failure", but there simply wasn't WG consensus for requiring anything stronger.